File: spec/models/post_spec.rb
RSpec.describe Post, type: :model do
    describe '#like_button' do
      it 'is white when post has no likes' do
        expect(subject.like_button).to eq('🤍')
      end
  
      it 'is red when post has >= 1 likes' do
        allow(subject).to receive(:likes) { [1] }
        expect(subject.like_button).to eq('❤️')
      end
    end
  end

File: spec/models/post.spec.js
var mongoose = require("mongoose");

require("../mongodb_helper");
var Post = require("../../models/post");

describe("Post model", () => {
  beforeEach((done) => {
    mongoose.connection.collections.posts.drop(() => {
      done();
    });
  });

  it("has a url", () => {
    var post = new Post({ url: "https://imgur.com/eI35OFg", caption: "Test" });
    expect(post.url).toEqual("https://imgur.com/eI35OFg");
  });

  it("can list all posts", (done) => {
    Post.find((err, posts) => {
      expect(err).toBeNull();
      expect(posts).toEqual([]);
      done();
    });
  });

  it("can save a post", (done) => {
    var post = new Post({ url: "https://imgur.com/eI35OFg", caption: "Test" });

    post.save((err) => {
      expect(err).toBeNull();

      Post.find((err, posts) => {
        expect(err).toBeNull();

        expect(posts[0]).toMatchObject({ caption: "Test" });
        done();
      });
    });
  });
});
